packages/lesswrong/lib/collections/users/schema.ts

```diff
@@ -185,16 +322,33 @@ const expandedFrontpageSectionsSettings = new SimpleSchema({
   popularComments: {type: Boolean, optional: true, nullable: true},
 });
 
-const notificationTypeSettingsField = (overrideSettings?: Partial<NotificationTypeSettings>) => ({
-  type: notificationTypeSettings,
-  optional: true,
-  group: formGroups.notifications,
-  control: "NotificationTypeSettings" as const,
-  canRead: [userOwns, 'admins'] as FieldPermissions,
-  canUpdate: [userOwns, 'admins'] as FieldPermissions,
-  canCreate: ['members', 'admins'] as FieldCreatePermissions,
-  ...schemaDefaultValue({ ...defaultNotificationTypeSettings, ...overrideSettings })
-});
+type DeepPartial<T> = {
+  [P in keyof T]?: T[P] extends object ? DeepPartial<T[P]> : T[P];
+};
+
+const notificationTypeSettingsField = (overrideSettings?: DeepPartial<NotificationTypeSettings>) => {
+  const defaultValue = {
+    onsite: { ...defaultNotificationTypeSettings.onsite, ...overrideSettings?.onsite },
+    email: { ...defaultNotificationTypeSettings.email, ...overrideSettings?.email }
+  }
+
+  // TODO remove once migration is complete
+  const migrationFields = {
+    blackbox: true
+  }
```

Comment:
Without this submitting the account settings form throws an error if a user still has some fields in the old format